build.gradle.kts

```diff
@@ -0,0 +1,71 @@
+import com.example.buildsrc.MyExtension
+
+plugins {
+    id("java")
+}
+
+group = "org.example"
+version = "1.0-SNAPSHOT"
+
+repositories {
+    mavenCentral()
+}
+
+dependencies {
+    testImplementation("org.junit.jupiter:junit-jupiter-api:5.8.1")
+    testRuntimeOnly("org.junit.jupiter:junit-jupiter-engine:5.8.1")
+}
+
+tasks.getByName<Test>("test") {
+    useJUnitPlatform()
+}
+
+allprojects{
+    extensions.create("Extension", MyExtension::class.java)
+
+    //passing a no-argument lambda makes the val initialized with lambda result
+    val addElementNoProjectAllScope by extra { project: Project ->
+        configure<MyExtension> {
+            contents.add("addElementNoProjectAllScope")
+        }
+    }
+
+    val addElementFakeLambdaAllScope by extra {
```

Comment:
Actually... why do you call this lambda _fake_? 🤔

Reply:
Mostly for memes, but I initially thought this val is a NO parameter lambda, but turns out it is NOT.

Reply:
> for memes

![image](https://user-images.githubusercontent.com/3383210/180019604-1e860c96-e759-4a00-bc9e-43eaf7b68bca.png)

Reply:
> it is NOT.

it is **not a lambda** at all, right?

Reply:
Unlike with groovy, where basically any loose piece of code in `{ ... }` is a lambda ("closure") ;p

Reply:
> Is it because it's really a call to `ExtraPropertiesExtension.invoke`? 🤔 pls put a comment here in code, preferably with a some desugaring of this construct into a more Java-like equivalent

Exactly, this function is chosen by the compiler and it looks like this: [invoke](https://gradle.github.io/kotlin-dsl-docs/api/org.gradle.kotlin.dsl/org.gradle.api.plugins.-extra-properties-extension/invoke.html) </br>

> Unlike with groovy, where basically any loose piece of code in `{ ... }` is a lambda ("closure") ;p

I think that also in Kotlin _"any loose piece of code in { ... } is a lambda"_ or at least can be interpreted as such.